### PR TITLE
Instruct gemini code assist not to comment on syntax error

### PR DESCRIPTION
**Pull request by chingjun**
Gemini doesn't know new Dart syntax, and is only adding noise in that case.

### CODE REVIEW

File: .gemini/styleguide.md [hunk 1]
```diff
@@ -29,6 +29,10 @@ flutter/flutter repository. It is based on the more comprehensive official
 - **Search for counter-examples**: Identify scenarios or edge cases that the proposed code does not handle. If a counter-example is found, propose a test case to demonstrate the gap.
 - **Suggest simplification and refactoring**: Assess whether the code can be made simpler or refactored to enhance readability and maintainability.
 
+### What Not to Report
```
Comment (Piinks) [inline .gemini/styleguide.md:32]: It should still report them if the analyzer does though, right? Should we see how to teach the agent about new Dart features?

Comment (chingjun) [reply .gemini/styleguide.md:32]: If the error would have been caught by another presubmit anyway, the comment is just adding noise. It's like having two tests that run the exact same test.

Another reason is future proof. While I agree that it would be better to teach the agent about new Dart features so that it knows when to use them (and we should definitely add that when the Dart team publishes the official skill for that), with this change at least it won't add noise on things that it has outdated information about.

**Review by jtmcdole — APPROVED** [review]
